sound/hda/ext/hdac_ext_stream.c

```diff
@@ -378,15 +386,17 @@ void snd_hdac_ext_stream_release(struct hdac_ext_stream *stream, int type)
 		break;
 
 	case HDAC_EXT_STREAM_TYPE_HOST:
+		spin_lock_irq(&bus->reg_lock);
 		if (stream->decoupled && !stream->link_locked)
-			snd_hdac_ext_stream_decouple(bus, stream, false);
+			snd_hdac_ext_stream_decouple_locked(bus, stream, false);
+		spin_unlock_irq(&bus->reg_lock);
 		snd_hdac_stream_release(&stream->hstream);
 		break;
 
 	case HDAC_EXT_STREAM_TYPE_LINK:
+		spin_lock_irq(&bus->reg_lock);
 		if (stream->decoupled && !stream->hstream.opened)
 			snd_hdac_ext_stream_decouple(bus, stream, false);
```

Comment:
snd_hdac_ext_stream_decouple_locked ?

Reply:
yes indeed, thanks for spotting this @RanderWang

Reply:
fixed.